src/utils_GUI/event_functions.jl

```diff
@@ -0,0 +1,468 @@
+"""
+    define_event_functions(gui::GUI)
+
+Define event functions (handling button clicks, plot updates, etc.) for the GUI `gui`.
+"""
+function define_event_functions(gui::GUI)
```

Comment:
The `define_event_functions` function initializes all event functions to the `gui` object, which must must be done once when calling the `GUI` constructor. I guess you can have several functions initializing different groups of event functions. But I'm not sure that improves the code quite frankly.

Reply:
It would be potentially easier to read based on the individual comments, _e.g._, `# Handle case when user is pressing/releasing any ctrl key (in order to select multiple components)` and the others following afterwards. In this case, you would define the event functions separately.

It will lead to more code lines, but could improve the readability. I leave it however to you

Reply:
One could skip wrapping theses event functions into a new function and simply include("utils_GUI/event_functions.jl") in the GUI constructor (instead of [calling](https://github.com/EnergyModelsX/EnergyModelsGUI.jl/blob/567d56464ccd73f8f6670a8b4f2e48eabc9bc999/src/setup_GUI.jl#L170) `define_event_functions(gui)`. Would that be better?